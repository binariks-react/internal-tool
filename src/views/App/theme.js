export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
