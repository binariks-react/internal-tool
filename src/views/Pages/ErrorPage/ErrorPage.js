import React from 'react';
import styled from 'styled-components';
import Button from 'views/Components/Controls/Button';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
`;

const ErrorCodeDiv = styled.div`
  height: 88px;
  font-family: Roboto;
  font-style: normal;
  font-weight: 900;
  font-size: 75px;
  line-height: normal;
  letter-spacing: -11.5385px;
  color: ${ props => props.theme.colors.darkGrey };
`;

const ErrorTitleDiv = styled.div`
  height: 59px;
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 50px;
  line-height: normal;
  letter-spacing: -5px;
  color: ${props => props.theme.colors.darkBlue};
`;

const ErrorMessageDiv = styled.div`
  height: 23.5px;
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 20px;
  line-height: normal;
  letter-spacing: -1px;
  color: ${props => props.theme.colors.secondary};
`;

const ErrorPage = ({ errorCode, errorTitle, errorMessage, goBack }) => (

  <Wrapper>
    <ErrorCodeDiv>{errorCode}</ErrorCodeDiv>
    <ErrorTitleDiv>{errorTitle}</ErrorTitleDiv>
    <ErrorMessageDiv>{errorMessage}</ErrorMessageDiv>
    <Button onClick={goBack} color="primary">Go Back</Button>
  </Wrapper>

);

const defaultProps = {
  errorCode: '500',
  errorTitle: 'Something went wrong',
  errorMessage: 'There was a problem on our end. Please try again later',
  goBack: () => {},
};

const propTypes = {
  errorCode: PropTypes.string,
  errorTitle: PropTypes.string,
  errorMessage: PropTypes.string,
  goBack: PropTypes.func,
};

ErrorPage.defaultProps= defaultProps;
ErrorPage.propTypes = propTypes;

export default ErrorPage;
