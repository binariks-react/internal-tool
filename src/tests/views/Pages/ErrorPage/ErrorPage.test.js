import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render, mount } from 'enzyme';
import theme from 'views/App/theme';
import ErrorPage from 'views/Pages/ErrorPage/ErrorPage';
import 'jest-styled-components';

describe('ErrorPage', () => {
  const goBack = jest.fn();
  beforeEach(() => {
    goBack.mockClear();
  });

  it('default', () => {
    const errorPage = render(
      <ThemeProvider theme={theme}>
        <ErrorPage />
      </ThemeProvider>
    );
    expect(errorPage).toMatchSnapshot();
  });
  it('check error Code', () => {
    const errorPage = render(
      <ThemeProvider theme={theme}>
        <ErrorPage errorCode="404" errorTitle="PAGE NOT FOUND" errorMessage="please check your route" />
      </ThemeProvider>
    );
    expect(errorPage).toMatchSnapshot();
  });
  it('go back function', () => {
    const errorPage = mount(
      <ThemeProvider theme={theme}>
        <ErrorPage goBack={goBack} />
      </ThemeProvider>
    );
    errorPage.find('button').simulate('click');
    expect(goBack).toHaveBeenCalledTimes(1);
  });
  it('goback default', () => {
    const errorPage = mount(
      <ThemeProvider theme={theme}>
        <ErrorPage />
      </ThemeProvider>
    );
    errorPage.find('button').simulate('click');
  });
});


