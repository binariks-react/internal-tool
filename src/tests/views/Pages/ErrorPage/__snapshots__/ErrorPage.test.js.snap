// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage check error Code 1`] = `
.c4 {
  background-color: #007bff;
  border-color: #007bff;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.125;
  padding: .5625rem 1rem;
  font-size: .75rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: #ffffff;
}

.c4:hover {
  background-color: #006fe6;
  border-color: #006fe6;
  box-shadow: 0 5px 15px rgba(0,0,0,.05),0 4px 10px rgba(0,123,255,.25);
  color: #ffffff;
}

.c4:active {
  background-color: #006fe6;
  border-color: #0062cc;
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

.c4:focus {
  box-shadow: 0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  width: 100%;
}

.c1 {
  height: 88px;
  font-family: Roboto;
  font-style: normal;
  font-weight: 900;
  font-size: 75px;
  line-height: normal;
  -webkit-letter-spacing: -11.5385px;
  -moz-letter-spacing: -11.5385px;
  -ms-letter-spacing: -11.5385px;
  letter-spacing: -11.5385px;
  color: #becad6;
}

.c2 {
  height: 59px;
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 50px;
  line-height: normal;
  -webkit-letter-spacing: -5px;
  -moz-letter-spacing: -5px;
  -ms-letter-spacing: -5px;
  letter-spacing: -5px;
  color: #3d5170;
}

.c3 {
  height: 23.5px;
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 20px;
  line-height: normal;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
  color: #5a6169;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    404
  </div>
  <div
    class="c2"
  >
    PAGE NOT FOUND
  </div>
  <div
    class="c3"
  >
    please check your route
  </div>
  <button
    class="c4"
    color="primary"
    fill="fill"
    type="button"
  >
    Go Back
  </button>
</div>
`;

exports[`ErrorPage default 1`] = `
.c4 {
  background-color: #007bff;
  border-color: #007bff;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.125;
  padding: .5625rem 1rem;
  font-size: .75rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: #ffffff;
}

.c4:hover {
  background-color: #006fe6;
  border-color: #006fe6;
  box-shadow: 0 5px 15px rgba(0,0,0,.05),0 4px 10px rgba(0,123,255,.25);
  color: #ffffff;
}

.c4:active {
  background-color: #006fe6;
  border-color: #0062cc;
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

.c4:focus {
  box-shadow: 0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  width: 100%;
}

.c1 {
  height: 88px;
  font-family: Roboto;
  font-style: normal;
  font-weight: 900;
  font-size: 75px;
  line-height: normal;
  -webkit-letter-spacing: -11.5385px;
  -moz-letter-spacing: -11.5385px;
  -ms-letter-spacing: -11.5385px;
  letter-spacing: -11.5385px;
  color: #becad6;
}

.c2 {
  height: 59px;
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 50px;
  line-height: normal;
  -webkit-letter-spacing: -5px;
  -moz-letter-spacing: -5px;
  -ms-letter-spacing: -5px;
  letter-spacing: -5px;
  color: #3d5170;
}

.c3 {
  height: 23.5px;
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 20px;
  line-height: normal;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
  color: #5a6169;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    500
  </div>
  <div
    class="c2"
  >
    Something went wrong
  </div>
  <div
    class="c3"
  >
    There was a problem on our end. Please try again later
  </div>
  <button
    class="c4"
    color="primary"
    fill="fill"
    type="button"
  >
    Go Back
  </button>
</div>
`;
